packages/cli-types/src/ios.ts

```diff
@@ -31,3 +38,55 @@ export interface IOSProjectConfig {
 }
 
 export interface IOSDependencyConfig extends IOSProjectConfig {}
+
+/**
+ * @see https://www.rubydoc.info/gems/cocoapods-core/Pod/Podfile/DSL#script_phase-instance_method
+ *
+ * The only difference is that `script` may be omitted in favour of a
+ * `path`, relative to the root of the package, whose content will be
+ * used.
+ */
+export type IOSScriptPhase = ({script: string} | {path: string}) & {
+  name: string;
+  shell_path?: string;
+  input_files?: string[];
+  output_files?: string[];
+  input_file_lists?: string[];
+  output_file_lists?: string[];
+  show_env_vars_in_log?: boolean;
+  dependency_file?: string;
+  execution_position?: 'before_compile' | 'after_compile' | 'any';
+};
+
+/**
+ * This describes the data that is expected by `native_modules.rb`. It is only
+ * meant to ensure the `Config` interface follows exactly what is needed, so
+ * only make changes to this interface (or `IOSScriptPhase`) if the data
+ * requirements of `native_modules.rb` change.
+ */
+export interface IOSNativeModulesConfig {
```

Comment:
❗ So this should only change if the Ruby source changes and should not be refactored/shared with other typings, even if they are similar.

Reply:
Why is that? Wouldn't it be better to just reference https://github.com/react-native-community/cli/blob/f570baf34072d3256064d92162333bd173c96c78/packages/cli-types/src/index.ts#L131-L153 directly? This is the exact type of the JSON output from the command.

Could you help me understand where and how it is used? Maybe it's not the right place for it then (in shared package), but next to a utility?

Reply:
Because TS cannot type-check the Ruby code, so people making changes to this `Config` interface might accidentally break some input data assumption made by it. Put differently, you should see this interface as if it were the function signature declaration of the Ruby code, if it were written in TS.